scripts/codeshift/convertToUseFragment.ts

```diff
@@ -0,0 +1,181 @@
+    // call expressions until we hit the identifier, or by taking in the component name from the
+    // command line or file name.
+    throw new Error('trying to convert component with non-trivial HOC wrapping')
+  }
+
+  const componentName = replacementExport.name
+
+  // Build a map of propName -> graphql fragment.
+  const fragmentProps = functionExpression.arguments[1]
+  const propMap = {}
+  fragmentProps.properties.forEach((prop) => {
+    propMap[prop.key.name] = prop.value
+  })
+
+  // Find the declaration of the component we're changing
+  const componentDeclaration = root.find(j.VariableDeclarator, {id: {name: componentName}})
+  if (componentDeclaration.size() !== 1) {
+    throw new Error("can't find component!")
+  }
+
+  // Find where the component destructures its props:
+  //  const {myProp} = props
+  const propDestructuring = componentDeclaration.find(j.VariableDeclaration, {
+    declarations: [
+      {
+        id: {type: 'ObjectPattern'},
+        init: {type: 'Identifier', name: 'props'}
+      }
+    ]
+  })
+
+  if (propDestructuring.size() !== 1) {
+    throw new Error('props destructured never or more than once')
+  }
+
+  // Check whether this is renaming any props when destructuring:
+  //  const {myProp: otherNameForProp} = props
+  const hasRenamedProps =
+    propDestructuring
+      .find(j.ObjectProperty)
+      .filter(
+        (p) =>
+          p.value.key.type === 'Identifier' &&
+          Object.keys(propMap).includes(p.value.key.name) &&
+          p.value.value.type === 'Identifier' &&
+          p.value.value.name !== p.value.key.name
+      )
+      .size() > 0
+
+  if (hasRenamedProps) {
+    // :TODO: (jmtaber129): Support this.
+    throw new Error('prop names renamed during prop destructuring')
+  }
+
+  // Rename the destructured props to '<propName>Ref'. From
+  //   const {myProp} = props
+  // to
+  //   const {myProp: myPropRef} = props
+  propDestructuring
+    .find(j.ObjectProperty)
+    .filter(
+      (p) =>
+        p.value.key.type === 'Identifier' &&
+        Object.keys(propMap).includes(p.value.key.name) &&
+        p.value.value.type === 'Identifier' &&
+        p.value.value.name === p.value.key.name
+    )
+    .replaceWith((p) => {
+      // Make the destructured name appended with Ref.
+      if (p.value.key.type === 'Identifier' && p.value.value.type === 'Identifier') {
+        return j.objectProperty.from({
+          key: p.value.key,
+          value: j.identifier(`${p.value.key.name}Ref`)
+        })
+      }
+    })
+
+  // Insert the 'useFragment' call right after the prop destructuring:
+  //   const {myProp: myPropRef} = props
+  //   const myProp = useFragment(graphql`<fragment>`, myPropRef)
+  propDestructuring.insertAfter(() => {
+    return Object.keys(propMap).map((prop) => {
+      console.log(prop)
+      return statement`const ${prop} = useFragment(${propMap[prop]}, ${j.identifier(
+        prop + 'Ref'
```

Comment:
Good catch! Added a check + error to catch that case.

I think we should maintain consistency for now, since devs might get confused when some things are called key and some are called ref. If there comes a point where we want to change to suffix `Key`, it shouldn't be _too_ difficult to automate it. I'll move forward with the suffix `Ref`